clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -689,6 +689,9 @@ private void handleListOffsetResponse(Map<TopicPartition, Long> timestampsToSear
                     parsedRecords = new PartitionRecords<>(fetchOffset, tp, parsed);
                     ConsumerRecord<K, V> record = parsed.get(parsed.size() - 1);
                     this.sensors.recordsFetchLag.record(partition.highWatermark - record.offset());
+                } else if (partition.records.iterator().hasNext()) {
```

Comment:
@hachikuji Yes, good point. I think we can compute lag as `partition.highWatermark - position` if parsed result is empty. But I am not sure if we should use this as lag if parsed result is not empty. It may be a good solution. But doing so changes the semantic of the fetch lag metric since the metric would only be updated when position is updated in the next `poll()` call.